agents-api/tests/test_tool_calls_evaluator.py

```diff
@@ -0,0 +1,1807 @@
+        },
+    ]
+)
+
+task_create = make_acompletion_multiple_outputs(
+    lambda agent, doc, user, task, user_doc, session: [
+        {
+            "role": "assistant",
+            "content": "Creating task",
+            "tool_calls": [
+                {
+                    "id": "call_task_create",
+                    "type": "system",
+                    "function": {
+                        "name": "task.create",
+                        "arguments": {
+                            "developer_id": str(UUID(int=0)),
+                            "agent_id": str(agent.id),
+                            "data": {"name": "Test Task", "main": [{"sleep": {"seconds": 1}}]},
+                        },
+                    },
+                }
+            ],
+        },
+        {
+            "role": "assistant",
+            "content": "Task created successfully",
+        },
+    ]
+)
+
+task_update = make_acompletion_multiple_outputs(
+    lambda agent, doc, user, task, user_doc, session: [
+        {
+            "role": "assistant",
+            "content": "Updating task",
+            "tool_calls": [
+                {
+                    "id": "call_task_update",
+                    "type": "system",
+                    "function": {
+                        "name": "task.update",
+                        "arguments": {
+                            "task_id": str(task.id),
+                            "agent_id": str(agent.id),
+                            "developer_id": str(UUID(int=0)),
+                            "data": {
+                                "name": "Updated Task",
+                                "main": [{"sleep": {"seconds": 1}}],
+                            },
+                        },
+                    },
+                }
+            ],
+        },
+        {
+            "role": "assistant",
+            "content": "Task updated successfully",
+        },
+    ]
+)
+
+task_delete = make_acompletion_multiple_outputs(
+    lambda agent, doc, user, task, user_doc, session: [
+        {
+            "role": "assistant",
+            "content": "Deleting task",
+            "tool_calls": [
+                {
+                    "id": "call_task_delete",
+                    "type": "system",
+                    "function": {
+                        "name": "task.delete",
+                        "arguments": {
+                            "task_id": str(task.id),
+                            "developer_id": str(UUID(int=0)),
+                        },
+                    },
+                }
+            ],
+        },
+        {
+            "role": "assistant",
+            "content": "Task deleted successfully",
+        },
+    ]
+)
+
+
+@test("chat: evaluate agent.doc.list tool call")
+async def _(
+    make_request=make_request,
+    developer_id=test_developer_id,
+    agent=test_agent,
+    mocks=agent_doc_list,
+    dsn=pg_dsn,
+):
+    pool = await create_db_pool(dsn=dsn)
```

Comment:
shouldn't the correct test be like:

```python
    test_agent = client.agents.create(..., tools=[dict(type='system', ...)])  # add agent.docs.list system tool
    # ...
    mocks = ... #  from test_agent.tools?
    
    response = client.sessions.chat(agent=test_agent, ...)  # this should call the mocks
    # other tests, asserts etc
```

Reply:
this involves the client, which might change interface in the future